src/scope.rs

```diff
@@ -286,20 +253,18 @@ impl<S: 'static> Scope<S> {
     where
         F: FnOnce(&mut Resource<S>) -> R + 'static,
     {
-        // add resource handler
-        let mut handler = Resource::default();
-        f(&mut handler);
-
+        // add resource
         let pattern = ResourceDef::with_prefix(
-            handler.get_name(),
             path,
             if path.is_empty() { "" } else { "/" },
             false,
         );
-        Rc::get_mut(&mut self.resources)
-            .expect("Can not use after configuration")
-            .push((pattern, Rc::new(handler)));
+        let mut resource = Resource::new(pattern);
+        f(&mut resource);
 
+        Rc::get_mut(&mut self.router)
+            .unwrap()
```

Comment:
I think it would be better to continue using `expect` with some message to easily identify problem, even if it is unlikely to happen.

Reply:
i can not imagine how this unwrap could panic

Reply:
This is true, but in case of any future modification that may lead to such situation...
Better to be easy to determine problem 😄 

But anyway it is minor thing so feel free to ignore

Reply:
you can add `except` if you'd like :)